prov/shm/src/smr_progress.c

```diff
@@ -513,11 +511,11 @@ static int smr_progress_ipc(struct smr_cmd *cmd, enum fi_hmem_iface iface,
 						     ptr, cmd->msg.hdr.size);
 	}
 
-	if (cmd->msg.data.ipc_info.iface == FI_HMEM_ZE)
+	if (cmd->msg.data.ipc_info.iface == FI_HMEM_ZE) {
 		close(ipc_fd);
-
-	/* Truncation error takes precedence over close_handle error */
-	ret = ofi_hmem_close_handle(cmd->msg.data.ipc_info.iface, base);
+		/* Truncation error takes precedence over close_handle error */
```

Comment:
Was this supposed to be encapsulated within the if block? Is this a bugfix or a bug?

Reply:
So before this PR, the IPC handle is closed every time after data copy (for both ZE and non-ZE). But with this PR, we do not want to close handle each time. ZE is an exception because this PR does not want to modify ZE's behavior, so the close is moved to the if block.

Reply:
@aingerson ZE behaves differently because of the use of ze_hmem_open_shared_handle(). Can ZE be adapted to take advantage of the IPC cache? for example, moving the call of ze_hmem_open_shared_handle() into the IPC cache code, or using the shared handle as the IPC handle? Can be done as a follow up PR.

Reply:
Sorry for some reason I didn't see this earlier. We can take advantage of the cache to a certain extent. The handles are fds so we need to make sure to take into consideration having too many fds open. But I think this can be done. I can follow up after this is merged.